guides/source/active_storage_overview.md

```diff
@@ -985,51 +1008,141 @@ class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
 end
 ```
 
+If you're using [parallel tests][] and the Disk service, you should configure each process to use its own
+folder for Active Storage. This way, the `teardown` callback will only delete files from the relevant process'
+tests.
+
+```ruby
+class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
+  # ...
+  parallelize_setup do |i|
+    ActiveStorage::Blob.service.root = "#{ActiveStorage::Blob.service.root}-#{i}"
+  end
+  # ...
+end
+```
+
 If your system tests verify the deletion of a model with attachments and you're
 using Active Job, set your test environment to use the inline queue adapter so
 the purge job is executed immediately rather at an unknown time in the future.
 
-You may also want to use a separate service definition for the test environment
-so your tests don't delete the files you create during development.
-
 ```ruby
 # Use inline job processing to make things happen immediately
 config.active_job.queue_adapter = :inline
-
-# Separate file storage in the test environment
-config.active_storage.service = :local_test
 ```
 
-Discarding Files Stored During Integration Tests
--------------------------------------------
+[parallel tests]: https://guides.rubyonrails.org/testing.html#parallel-testing
+
+#### Integration tests
 
 Similarly to System Tests, files uploaded during Integration Tests will not be
 automatically cleaned up. If you want to clear the files, you can do it in an
-`after_teardown` callback. Doing it here ensures that all connections created
-during the test are complete and you won't receive an error from Active Storage
-saying it can't find a file.
+`teardown` callback.
 
 ```ruby
-module RemoveUploadedFiles
-  def after_teardown
-    super
-    remove_uploaded_files
+module ActionDispatch
+  class IntegrationTest
+    teardown do
```

Comment:
This should probably continue be `after_teardown` in case a user defines a teardown block that references an uploaded file. Also I think the example above uses `def after_teardown` when it could be using `after_teardown do` like this one.

Reply:
Will update. Re the block - `after_teardown` doesn't take one: https://github.com/rails/rails/blob/main/activesupport/lib/active_support/testing/setup_and_teardown.rb#L44

So the correct pattern seems to be to override the method and call `super`.